rns_shell/compositor/layers/Layer.cpp

```diff
@@ -105,17 +105,13 @@ void Layer::removeFromParent() {
 
 void Layer::preRoll(PaintContext& context, bool forceLayout) {
     // Layer Layout has changed or parent has forced Layout change for child. Need to recalculate absolute frame and update absFrame_
-    if(forceLayout || (invalidateMask_ & LayerLayoutInvalidate)) {
-        // Adjust absolute position
-        //TODO Once we have skia transform matrix, need to first update frame_ i.e before calculating newAbsFrame
-        // layer_->setFrame( trnsfromed x, y width hight
-        int frameAbsX = frame_.x(), frameAbsY = frame_.y();
-        if(parent_) {
-            frameAbsX += parent_->absFrame_.x();
-            frameAbsY += parent_->absFrame_.y();
-        }
+     if(forceLayout || (invalidateMask_ & LayerLayoutInvalidate)) {
+        // Adjust absolute position w.r.t transformation matrix
+        calculateTransformMatrix();
+        SkRect mapRect=SkRect::Make(frame_);
+        absoluteTransformMatrix_.mapRect(&mapRect);
 
-        SkIRect newAbsFrame = SkIRect::MakeXYWH(frameAbsX, frameAbsY, frame_.width(), frame_.height());
+        SkIRect newAbsFrame = SkIRect::MakeXYWH(mapRect.x(), mapRect.y(), mapRect.width(), mapRect.height());
 #if USE(RNS_SHELL_PARTIAL_UPDATES)
         if((invalidateMask_ & LayerLayoutInvalidate)) {
```

Comment:
Though not related to this PR, there is a redundant check here. And also, need to check context.supportPartialUpdate whenever we want to add damage rect.